modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,196 @@
++
+To create and expose other types of services externally, outside of the cluster,
+such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate,
+in the Service object definition.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <3>
+  selector:
+    special: key <4>
+  type: ClusterIP <5>
+----
+<1> Service name
+<2> `namespace` corresponds to the `namespace` you referenced in the virtual machine YAML.
+<3> Exposes SSH `targetPort` 22
+<4> `selector` `special: key` corresponds to the `labels` you added in the virtual machine YAML.
+<5> Defines Service `type: ClusterIP`
+
++
+. Create the `ClusterIP` service using `kubectl`.
++
+
+[source,terminal]
+----
+$ kubectl create -f <service_name>.yaml
+----
+
++
+. Start the virtual machine. If the virtual machine is already running, restart it.
++
+
++
+. Query the Service object to verify it is available and is configured with type `ClusterIP`.
++
+
+[source, terminal]
+----
+$ kubectl get service
+----
++
+
+.Example output
+[source, terminal]
+----
+NAME        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)     AGE
+vmservice  ClusterIP   172.30.3.149   <none>        27017/TCP   2m
+----
++
+
+. Connect to the `VirtualMachine` using the service IP and service port.
+* Exposing a virtual machine's SSH port through a `ClusterIP` service enables
+connection to a virtual machine within the cluster, either through
+another pod or virtual machine.
+* `NodePort` and `LoadBalancer` services enable connection
+from outside of a cluster, as long as you can tunnel
+traffic to the cluster's node IP addresses.
+.. List all information
+about a project’s pod deployment by running this command, supplying
+node name:
++
+
+[source,terminal]
+----
+$ oc get node <node_name> -o wide
+----
++
+
+.Example output
+[source,terminal]
+----
+NAME                           STATUS    ROLES     AGE       VERSION           INTERNAL-IP      OS-IMAGE                            KERNEL-VERSION          CONTAINER-RUNTIME
+ip-172-18-0-199.ec2.internal   Ready     infra     1d        v1.10.0+b81c8f8   54.172.185.130   Fedora cloud container disk demo    3.10.0-862.el7.x86_64   docker://1.13.1
+----
++
+
+.. Connect using the service IP and service port. In the `ssh` command, replace
+`<node_IP>` with the IP address in the INTERNAL-IP column from the previous `get node`
+command output. In this example, the `INTERNAL-IP` is `54.172.185.130`.
++
+
+[source,terminal]
+----
+$ ssh <user_name>@<node_IP> -p 27017
+----
+
+[NOTE]
+====
+* For type `LoadBalancer`, use the `vinagre` client to connect your
+`VirtualMachine` by using the public IP and port.
+External ports are dynamically allocated when using service type
+`LoadBalancer`. See the Additional Resources section "Configuring external IPs"
```

Comment:
I'd avoid implicit dependencies like this, pointing to the Additional resources section. This module could potentially be reused in another assembly that doesn't have that link.

I'd just remove it and let them find the link, which is right below anyway.

Reply:
Deleted